core/store/src/opener.rs

```diff
@@ -596,19 +595,19 @@ pub fn checkpoint_hot_storage_and_cleanup_columns(
 
     hot_store
         .storage
-        .create_checkpoint(&checkpoint_path)
+        .create_checkpoint(&checkpoint_path, columns_to_keep)
         .map_err(StoreOpenerError::CheckpointError)?;
 
     // As only path from config is used in StoreOpener, default config with custom path will do.
     let mut config = StoreConfig::default();
     config.path = Some(checkpoint_path);
     let archive = hot_store.get_db_kind()? == Some(DbKind::Archive);
     let opener = StoreOpener::new(checkpoint_base_path, archive, &config, None);
+    // This will create all the column families that were dropped by create_checkpoint(),
+    // but all the data and associated files that were in them previously should be gone.
     let node_storage = opener.open_in_mode(Mode::ReadWriteExisting)?;
 
-    if let Some(columns_to_keep) = columns_to_keep {
-        let columns_to_keep_set: std::collections::HashSet<DBCol> =
-            std::collections::HashSet::from_iter(columns_to_keep.into_iter());
+    if columns_to_keep.is_some() {
         let mut transaction = DBTransaction::new();
         // Force the checkpoint to be a Hot DB kind to simplify opening the snapshots later.
```

Comment:
Ah I guess ignore my previous comment. Perhaps we should add a third temperature for snapshots to make it less confusing and properly checked but that's not needed here for sure.

Reply:
yeah as you said in the other comment, it looks kind of sus to me, and I'm not 100% sure why it was added before, but maybe it's fine to keep it for now